tarantool/dbapi.py

```diff
@@ -0,0 +1,251 @@
+
+    @property
+    def rowcount(self):
+        """
+        This read-only attribute specifies the number of rows that the last
+        .execute*() produced (for DQL statements like SELECT) or affected (
+        for DML statements like UPDATE or INSERT).
+        """
+        return self._rowcount
+
+    def _check_result_set(self, error=None):
+        """
+        Closed method for raising an error when Cursor object does not have
+        any row to fetch. Useful for checking access after DQL requests.
+        """
+        if self._rows is None:
+            raise InterfaceError(error or "No result set to fetch from")
+
+    def fetchone(self):
+        """
+        Fetch the next row of a query result set, returning a single
+        sequence, or None when no more data is available.
+        """
+        self._check_result_set()
+        return self.fetchmany(1)[0] if self._rows else None
+
+    def fetchmany(self, size=None):
+        """
+        Fetch the next set of rows of a query result, returning a sequence of
+        sequences (e.g. a list of tuples). An empty sequence is returned when
+        no more rows are available.
+        """
+        self._check_result_set()
+
+        size = size or self.arraysize
+
+        if len(self._rows) < size:
+            items = self._rows
+            self._rows = []
+        else:
+            items, self._rows = self._rows[:size], self._rows[size:]
+
+        return items
+
+    def fetchall(self):
+        """Fetch all (remaining) rows of a query result, returning them as a
+        sequence of sequences (e.g. a list of tuples). Note that the cursor's
+        arraysize attribute can affect the performance of this operation.
+        """
+        self._check_result_set()
+
+        items = self._rows
+        self._rows = []
+        return items
+
+    def setinputsizes(self, sizes):
+        """PEP-249 allows to not implement this method and do nothing."""
+
+    def setoutputsize(self, size, column=None):
+        """PEP-249 allows to not implement this method and do nothing."""
+
+
+class Connection(BaseConnection):
+
+    def __init__(self, *args, **kwargs):
+        super(Connection, self).__init__(*args, **kwargs)
+        self._autocommit = kwargs.get('autocommit')
+
+    @property
+    def autocommit(self):
+        """Autocommit is True by default and the default will be changed
+        to False. Set the property explicitly to True or verify
+        it when lean on autocommit behaviour."""
+        return self._autocommit
+
+    @autocommit.setter
+    def autocommit(self, autocommit):
+        """Set autocommit state"""
+        if not isinstance(autocommit, bool):
+            raise InterfaceError("autocommit parameter must be boolean, "
+                                 "not %s" % type(autocommit))
+        if autocommit is False:
+            raise NotSupportedError("The connector supports "
+                                    "only autocommit mode")
+
```

Comment:
Nit: I would add `self._autocommit = autocommit` here just to don't forget about this, when we'll support interactive transactions. Or, maybe, just because it better fit to a template for a setter in my mind: verification and the actual assignment.

Reply:
Totally agree. Added this line.